handler/oauth2/flow_refresh.go

```diff
@@ -96,11 +96,20 @@ func (c *RefreshTokenGrantHandler) HandleTokenEndpointRequest(ctx context.Contex
 	}
 
 	request.SetSession(originalRequest.GetSession().Clone())
-	request.SetRequestedScopes(originalRequest.GetRequestedScopes())
 	request.SetRequestedAudience(originalRequest.GetRequestedAudience())
 
-	for _, scope := range originalRequest.GetGrantedScopes() {
-		if !c.Config.GetScopeStrategy(ctx)(request.GetClient().GetScopes(), scope) {
+	if _, ok := request.GetRequestForm()["scope"]; ok {
+		requestedScopes := fosite.RemoveEmpty(strings.Split(request.GetRequestForm().Get("scope"), " "))
```

Comment:
We need to check this against originalRequest.GetRequestedScopes() to ensure none of the scopes in the form `scope` value were not originally granted:

```
         The scope of the access request as described by Section 3.3.  
         The requested scope MUST NOT include any scope
         not originally granted by the resource owner, and if omitted is
         treated as equal to the scope originally granted by the
         resource owner.
```

Reply:
I don't think the originally **requested** scopes should be used anywhere in the refresh token handler. Note that your RFC quote above only mentions the originally **granted** scopes, not the originally **requested** scopes. 

I've ensured that all scopes in the form `scope` were originally granted by comparing them against the originally **granted** scopes list:
```
	for _, scope := range request.GetRequestedScopes() {
		if !c.Config.GetScopeStrategy(ctx)(originalRequest.GetGrantedScopes(), scope) {
```

Reply:
Oh I misread the change, the other review is more pertinent my bad.